core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1005,6 +1012,31 @@ public long getWorkerId(WorkerNetAddress workerNetAddress) {
     return workerId;
   }
 
+  @Override
+  public GetWorkerIdPResponse getWorkerId(WorkerNetAddress workerNetAddress,
+      String workerClusterId, int blocksNum) throws IOException {
+    // todo The return type may be the Alluxio class(such as WorkerMetaInfo)
+    //  instead of the GRPC class ?
+    LOG.info("worker {} Register, workerClusterId {}, blocksNum {}",
+        workerNetAddress.getHost(), workerClusterId, blocksNum);
+
+    String curClusterId = getClusterId();
+    RegisterCommandType commandType =
+        checkWorker(curClusterId, workerClusterId, blocksNum);
+    long workerId = getWorkerId(workerNetAddress);
+    GetWorkerIdPResponse response = GetWorkerIdPResponse.newBuilder()
+        .setRegisterCommandType(commandType)
+        .setClusterId(curClusterId)
+        .setWorkerId(workerId)
+        .build();
+
+    LOG.info("worker {} ,Register result: RegisterCommand {}, workerId {},"
+            + " current clusterId {}",
+        workerNetAddress.getHost(), response.getRegisterCommandType(),
+        response.getWorkerId(), response.getClusterId());
```

Comment:
```suggestion
    LOG.info("Allocated ID {} to worker {} with command {}",
        response.getWorkerId(), 
        workerNetAddress.getHost(), 
        response.getRegisterCommandType());
```
And you can add another log in your `checkWorker()` like when worker and master cluster ID don't match:
```
LOG.warn("Worker {} has a different clusterID {} from master {}. The worker may be outdated and the master command is {}, workerHostname, workerClusterId, masterClusterId, commandType);
```